posthog/permissions.py

```diff
@@ -153,3 +153,98 @@ def has_object_permission(self, request: Request, view, object: Model) -> bool:
             OrganizationMembership.objects.get(user=cast(User, request.user), organization=organization).level
             >= OrganizationMembership.Level.ADMIN
         )
+
+
+def get_effective_level(team: Team, user: User) -> Optional[OrganizationMembership.Level]:
```

Comment:
This method feels like repeated code from `models/team.py`, because it's particularly sensitive and can be complex, I think it would be better to simplify and keep in a single place.